python/dgl/contrib/sampling/sampler.py

```diff
@@ -406,6 +406,137 @@ def fetch(self, current_nodeflow_index):
         nflows = [NodeFlow(self.g, hdl) for hdl in handles]
         return nflows
 
+
+class EdgeSampler(object):
+    '''Base class that generates NodeFlows for link prediction.
+
+    For a mini-batch, it samples a set of edges. From the source and
+    destination vertices, it creates NodeFlows with node-level sampler
+    (e.g., NeighborSampler, Layer-wise sampler).
```

Comment:
It's not clear what this edge sampler could achieve. Could you give an example of the input and the possible output?